src/tests/detect-tls-cert.c

```diff
@@ -0,0 +1,363 @@
+/* Copyright (C) 2019 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Mats Klepsland <mats.klepsland@gmail.com>
+ *
+ */
+
+/**
+ * \test Test that a signature containing tls.cert is correctly parsed
+ *       and that the keyword is registred.
+ */
+static int DetectTlsCertTest01(void)
+{
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
+
+    de_ctx->flags |= DE_QUIET;
+    de_ctx->sig_list = SigInit(de_ctx, "alert tls any any -> any any "
+                               "(msg:\"Testing tls.cert\"; tls.cert; "
+                               "content:\"|01 02 03 04 05|\"; sid:1;)");
+    FAIL_IF_NULL(de_ctx->sig_list);
+
+    /* sm should not be in the MATCH list */
+    SigMatch *sm = de_ctx->sig_list->sm_lists[DETECT_SM_LIST_MATCH];
+    FAIL_IF_NOT_NULL(sm);
+
+    sm = de_ctx->sig_list->sm_lists[g_tls_cert_buffer_id];
+    FAIL_IF_NULL(sm);
+
+    FAIL_IF(sm->type != DETECT_CONTENT);
+    FAIL_IF_NOT_NULL(sm->next);
+
+    SigCleanSignatures(de_ctx);
```

Comment:
this is redundant: called by DetectEngineCtxFree

Reply:
I'll remove it!